Change disconnected upgrade steps (#4164)
theforeman/foreman-documentation: guides/common/modules/proc_upgrading-a-disconnected-project-server.adoc

----
# rm /etc/yum.repos.d/*
----

> .Procedure on the connected {ProjectServer}
> . Ensure that the following repositories are synchronized on your connected {ProjectServer}:

Reviewer: This would introduce a custom block title (`.Procedure on the connected...`) and those are not supported for the DITA migration. It's better not to introduce new ones. My suggestion for an alternative:

```suggestion
. Export the required repositories from your connected {ProjectServer} to your disconnected {ProjectServer}:
.. Ensure that the following repositories are synchronized on your connected server:
```

+
* {RepoRHEL9BaseOS}
* {RepoRHEL9AppStream}
* {RepoRHEL9ServerSatelliteServerProjectVersion}

> . Export the repositories in a syncable format:

Reviewer: ```suggestion
.. Export the repositories in a syncable format:
```

+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
$ hammer content-export complete repository \
--id=_Repo_ID_ --format=syncable
----

> . Copy the exported directories to the disconnected {ProjectServer}.

Reviewer: ```suggestion
.. Copy the exported directories to the disconnected {ProjectServer}.
```

> .Upgrade disconnected {ProjectServer}

Reviewer: ```suggestion
```

Reboot these hosts after the upgrade has completed.

. Remove old repositories:
+
[options="nowrap" subs="attributes"]
----
# rm /etc/yum.repos.d/*
----

. Identify the location of the copied exported directories from the syncable export on the connected {ProjectServer}.
. Create the `/etc/yum.repos.d/upgrade.repo` and update the `baseurl` directive accordingly:
+
[options="nowrap" subs="+quotes,attributes"]
----
[{RepoRHEL9BaseOS}]
name=Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)
metadata_expire=-1
gpgcheck=0
cost=500
baseurl=file:///<export baseos location>/content/dist/rhel9/9/x86_64/baseos/os
[{RepoRHEL9AppStream}]
name=Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)
mediaid=None
metadata_expire=-1
gpgcheck=0
cost=500
baseurl=file:///<export appstream location>/content/dist/rhel9/9/x86_64/appstream/os
[{Project}]
name={Project}
mediaid=None
metadata_expire=-1
gpgcheck=0
cost=500
baseurl=file:///<export location>/content/dist/layered/rhel9/x86_64/{project-context}/{ProjectVersion}/os/
----
+
. Upgrade {foreman-maintain} to the next version:
+
[options="nowrap" subs="attributes"]
----
# {foreman-maintain} self-upgrade --maintenance-repo-label={Project}

> ----

Reviewer: This looks like a mix of old steps from previous versions that sneaked in here again and lines that you're also adding below. I believe you should delete all of this.

----
# rm /etc/yum.repos.d/*
----
. Enable access to the exported content by configuring a repository:

> .. Identify the location of the copied exported directories from the syncable export on the connected server.

Author: Perhaps in the prerequisites, I could add:
On the connected {ProjectServer}, create a syncable export of the repository that you want to use for the upgrade. For more information, see https://content-preview.docs.redhat.com/en/documentation/red_hat_satellite/6.18/html-single/managing_content/index#exporting-the-library-environment-incrementally_content-management.

> .. Identify the location of the copied exported directories from the syncable export on the connected server.

Reviewer 2: I am confused as to why we didnt just replicate https://raw.githubusercontent.com/bangelic/foreman-documentation/a28c699e52cbb1836e207926c0769171e60748b5/guides/common/modules/proc_upgrading-a-disconnected-project-server.adoc verbatim as in what we had for 3725

There was a section exactly for this

```
.Procedure on the connected {ProjectServer}
. Ensure that the following repositories are synchronized on your connected {ProjectServer}:
+
* {RepoRHEL9BaseOS}
* {RepoRHEL9AppStream}
* {RepoRHEL9ServerSatelliteServerProjectVersion}
. Export the repositories in a syncable format:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
$ hammer content-export complete repository \
--id=_Repo_ID_ --format=syncable
----
. Copy the exported directories to the disconnected {ProjectServer}.
```

----
# rm /etc/yum.repos.d/*
----

. Export the required repositories from your connected {ProjectServer} to your disconnected {ProjectServer}:
.. Ensure that the following repositories are synchronized on your connected server:
+
* {RepoRHEL9BaseOS}
* {RepoRHEL9AppStream}
* {RepoRHEL9ServerSatelliteServerProjectVersion}
.. Export the repositories in a syncable format:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
$ hammer content-export complete repository \
--id=_Repo_ID_ --format=syncable
----
.. Copy the exported directories to the disconnected {ProjectServer}.

. Stop all {Project} services:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
# {foreman-maintain} service stop
----
. Enable access to the exported content by configuring a repository:
.. Identify the location of the copied exported directories from the syncable export on the connected server.

> .. Create the `/etc/yum.repos.d/upgrade.repo` file and update the `baseurl` directive to point to that location:

Reviewer 2: The section reads as below with No mention of it being run on the Disconnected server. 
```
. Stop all {Project} services:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
# {foreman-maintain} service stop
----
. Enable access to the exported content by configuring a repository:
.. Identify the location of the copied exported directories from the syncable export on the connected server.
.. Create the `/etc/yum.repos.d/upgrade.repo` file and update the `baseurl` directive to point to that location:
```
General theme should be imo

1.  On the connected server export repositories as syncable and copy them over to the disconnected
2. On the disconnected server stop all services, setup the upgrade repo and upgrade.

----
# rm /etc/yum.repos.d/*
----

. Export the required repositories from your connected {ProjectServer} to your disconnected {ProjectServer}:
.. Ensure that the following repositories are synchronized on your connected server:
+
* {RepoRHEL9BaseOS}
* {RepoRHEL9AppStream}
* {RepoRHEL9ServerSatelliteServerProjectVersion}
.. Export the repositories in a syncable format:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
$ hammer content-export complete repository \

> --id=_Repo_ID_ --format=syncable

Reviewer 3: To export the repository, users must have synchronized it before with download policy "immediate".

Also, I suggest to add a command to get the Repo IDs and explain that you need to run this command three times: once for each repo listed in lines 51-53.

----
.. Copy the exported directories to the disconnected {ProjectServer}.

. On the connected server, export the required repositories in a syncable format and copy the exported directories to the disconnected server.
. On the disconnected server, stop all {Project} services:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
# {foreman-maintain} service stop
----
. On the disconnected server, enable access to the exported content by configuring a repository:
.. Identify the location of the copied exported directories from the syncable export on the connected server.
.. Create the `/etc/yum.repos.d/upgrade.repo` file and update the `baseurl` directive to point to that location:
+

> [options="nowrap" subs="attributes"]

Reviewer 3: ```suggestion
[source, none, options="nowrap" subs="attributes"]
```

----

> # [{RepoRHEL9BaseOS}]

Reviewer 3: ```suggestion
[{RepoRHEL9BaseOS}]
```

You can have comments in `.repo` files, but each repo needs an ID. -> You should drop the leading `# ` in line 76, 82, and 89.

Example:

````
$ podman run --rm -it registry.access.redhat.com/ubi10/ubi:10.0 cat /etc/yum.repos.d/ubi.repo | head -n 6
[ubi-10-for-$basearch-baseos-rpms]
name = Red Hat Universal Base Image 10 (RPMs) - BaseOS
baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi10/10/$basearch/baseos/os
enabled = 1
gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
gpgcheck = 1
````

FYI: If you'd drop lines starting with `[`, `dnf makecache` will return an error because it cannot read the file anymore.

+
[options="nowrap" subs="+quotes"]
----
$ hammer repository list \
--organization-label _My_Organization_Label_
----
+
Run this command once for each of the three repositories listed above.
.. Ensure that the repositories are synchronized on your connected server with the download policy set to `immediate`.
.. Export each repository in a syncable format by using its repository ID:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
$ hammer content-export complete repository \
--id=_Repo_ID_ --format=syncable
----
.. Copy the exported directories to the disconnected {ProjectServer}.

. On the disconnected server, stop all {Project} services:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
# {foreman-maintain} service stop
----
. On the disconnected server, enable access to the exported content by configuring a repository:
.. Identify the location of the copied exported directories from the syncable export on the connected server.
.. Create the `/etc/yum.repos.d/upgrade.repo` file and update the `baseurl` directive to point to that location:
+
[source, none, options="nowrap" subs="attributes"]
----
[{RepoRHEL9BaseOS}]

> --name=Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs) \

Reviewer 4: @parthaa The double hyphens `--` at the beginning don't look like [allowed yum config syntax](https://linux.die.net/man/5/yum.conf). Has this been tested? Should I fix this?

Reviewer 4: Fixing here> https://github.com/theforeman/foreman-documentation/pull/4388

.. Identify the location of the copied exported directories from the syncable export on the connected server.
.. Create the `/etc/yum.repos.d/upgrade.repo` file and update the `baseurl` directive to point to that location:
+
[options="nowrap" subs="attributes"]
----
# [{RepoRHEL9BaseOS}]
--name=Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs) \
--metadata_expire=-1 \
--gpgcheck=0 \
--cost=500 \
--baseurl=file:///<export baseos location>/content/dist/rhel9/9/x86_64/baseos/os \
# [{RepoRHEL9AppStream}]
--name=Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs) \
--mediaid=None \
--metadata_expire=-1 \
--gpgcheck=0 \
--cost=500 \
--baseurl=file:///<export appstream location>/content/dist/rhel9/9/x86_64/appstream/os \
# [{Project}]
--name={Project} \
--mediaid=None \
--metadata_expire=-1 \
--gpgcheck=0 \
--cost=500 \
--baseurl=file:///<export location>/content/dist/layered/rhel8/x86_64/{project-context}/{ProjectVersion}/os/
----

. Upgrade the {foreman-maintain} utility to its latest version:
+
[options="nowrap" subs="attributes"]
----
# {foreman-maintain} self-upgrade  --maintenance-repo-label={Project}
----

> 

Author: Blank lines not necessary in either spot. Fixing them.
